src/Pirouette/Term/ToTla.hs

```diff
@@ -238,10 +276,12 @@ termToSpec opts sortedNames mainFun t = flip evalStateT tlaState0 $ flip runRead
                         >> throwError' (PEOther "CTreeIsNotAMatch")
   let next = tlaOpDef (tlaIdent "Next") [] $ tlaOr $ map (tlaIdentPrefixed "Wrapped") $ ctreeFirstMatches mctree
 
+  applies <- genApplies
+
   tlaPure $ logDebug "Assembling the spec"
   skel <- asks toSkeleton
   return $ wrapSpec skel
-         $ concat deps ++ defs ++ [next]
+         $ concat deps ++ applies ++ defs ++ [next]
```

Comment:
Since, as we discussed, TLA+ is sensitive to declarations order, and definitions and `applyN` functions can depend on each other, I'm not sure about the immediately obviously correct ordering.

Reply:
Since `applyN` are not higher-order, they can be mutually recursive:
```tla
RECURSIVE apply1(_,_)
RECURSIVE apply2(_,_,_)
...
RECURSIVE applyN(_,_,_,_, ... ,_)

apply1(f,x) == ...
apply2(f,x,y) == ...
...
applyN(f,x,y,...) == ...
```